src/dependency-manager/src/component-config/base.ts

```diff
@@ -20,9 +20,12 @@ export abstract class ComponentConfig extends BaseSpec {
   abstract getKeywords(): string[];
   abstract getAuthor(): string;
   abstract getParameters(): Dictionary<ParameterDefinitionSpec>;
```

Comment:
Worth adding convenience `getParameter(key: string)` methods to all these that return Dictionaries?

Reply:
I'm not sure this is any better than getParameters()[key]

Reply:
Fair enough as long as getParameters() never returns null or undefined.